prow/config/secrets_agent.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"os"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// SecretAgent watches a path and automatically loads the config stored
+// therein.
+type SecretAgent struct {
+	sync.Mutex
+	secretsMap map[string][]byte
+}
+
+func NewSecretAgent() *SecretAgent {
+	return &SecretAgent{
+		secretsMap: make(map[string][]byte),
+	}
+}
+
+// Start will begin polling the config file at the path. If the first load
+// fails, Start with return the error and abort. Future load failures will log
+// the failure message but continue attempting to load.
+func (sa *SecretAgent) Start(paths []string) error {
+	secretsMap, err := LoadSecrets(paths)
+	if err != nil {
+		return err
+	}
+
+	sa.secretsMap = secretsMap
+
+	go func() {
```

Comment:
Why not `fsnotify`? Have we added that watch to the config agent, @BenTheElder

Reply:
+1. fsnotify should be the best solution. In secretAgent should be easier to implement than the config agent. I will check that out.

Reply:
follow-up issue even though I don't feel strong about fsnotify